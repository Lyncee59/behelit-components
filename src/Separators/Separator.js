import React from 'react'
import styled from 'styled-components'

import { theme } from '../Tools/interpolation'

const Wrapper = styled.div`
  width: 100%;
  height: 2px;
  background-color: ${theme('separatorColor')};
`

const Separator = (props) => <Wrapper {...props} />

export default Separator
