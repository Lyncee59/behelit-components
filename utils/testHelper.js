import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)