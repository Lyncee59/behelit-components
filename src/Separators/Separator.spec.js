import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import Separator from './Separator'

describe('Separator', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<Separator />)

    expect(tree.find(Separator)).toMatchSnapshot()
  })
})
