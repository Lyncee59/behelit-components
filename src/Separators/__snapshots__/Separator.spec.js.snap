// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator should render correctly 1`] = `
.c0 {
  width: 100%;
  height: 2px;
}

<Separator>
  <styled.div>
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  width: 100%;
  height: 2px;
  background-color: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      />
    </StyledComponent>
  </styled.div>
</Separator>
`;
